jdaviz/cli.py

```diff
@@ -41,8 +47,7 @@ def main(filename, layout='default'):
     nbdir = tempfile.mkdtemp()
 
     with open(os.path.join(nbdir, 'notebook.ipynb'), 'w') as nbf:
-        nbf.write(notebook_template.replace('CONFIG_NAME', layout).replace(
-            'DATA_FILENAME', filename).strip())
+        nbf.write(notebook_template.replace('DATA_FILENAME', str(filepath).encode('unicode_escape').decode()).strip())
```

Comment:
This change did not work for me.  To understand why I think this might be illuminating:  https://stackoverflow.com/a/24519338/534241 

The following, however *did* work:
```suggestion
        nbf.write(notebook_template.replace('DATA_FILENAME', str(filepath).replace('\\', '/')).strip())
```

I have no idea why.  This really *shouldn't* work, but I speculate there's some sort of magic path translation happening on the glue side?

An alternative fix might be to change the template to have it end up as ``r'filename\whatever'`` instead of ``'filename\whatever'``.  But that did *not* work for me when I mocked it up, so I think my current solution is probably better for now.

Reply:
Ooh sidestepping the problem entirely! I think the reason this works is because I added a line in the base application class' load_data method that processes the string given using `pathlib.Path`, which corrects the given text to the native system path format.
```
        try:
            # Properly form path and check if a valid file
            file_obj = pathlib.Path(file_obj)
            if not file_obj.exists():
                raise FileNotFoundError("Could not locate file: " + file_obj)
            else:
                # Convert path to properly formatted string (Parsers do not accept path objs)
                file_obj = str(file_obj)
```
My best understanding is this edit changes any windows "\\" characters to the unix "/" ones, regardless of platform, and then pathlib rectifies it back to "\\" but with the correct escape characters.

I checked this change on my side, and all three tools seem to accept the change